### PR TITLE
Removed warnings

### PR DESCRIPTION
**Pull request by EECOLOR**
Removed about 150 warnings (mainly in the compiler and reflect) project.

I am very unfamiliar with the Scala compiler so a review is probably in order.
- Added `since` to deprecation statement
- Added unit to parameter list
- Removed usage of deprecated method polyType
- Replaced deprecated `isPackage` with `hasPackageFlag`
- Replaced deprecated `getter` with `getterIn`
- Replaced deprecated `debugwarn` with `devWarning`
- Changed switch statement to if else in order to remove a warning
- Replaced deprecated `hasSymbol` with `hasSymbolField`
- Replaced deprecated `setter` with `setterIn`
- Replaced deprecated `newTermName` with `TermName`
- Switched implementation of `init` and `processOptions` to prevent
  warning
- Replaced deprecated `originalName` with `unexpandedName`
- Replaced deprecated `Console.readLine` with `scala.io.StdIn.readLine`
- Replaced deprecated `startOrPoint` with `start`
- Replaced deprecated `stringToTermName` with `TermName`
- Replaced deprecated `stripModuleSuffix` with `dropModule`
- Replaced deprecated `superSymbol` with `superSymbolIn`
- Replaced deprecated `tpe_=` with `setType`
- Replaced deprecated `typeCheck` with `typecheck`
- Replaced deprecated `CompilationUnit.warning` with `typer.context.warning`
- Replaced deprecated `scala.tools.nsc.util.ScalaClassLoader` with `scala.reflect.internal.util.ScalaClassLoader`
- Replaced deprecated `scala.tools.ListOfNil` with `scala.reflect.internal.util.ListOfNil`
- Replaced deprecated `scala.tools.utils.ScalaClassLoader` with `scala.reflect.internal.util.ScalaClassLoader`
- Replaced deprecated `tpnme` with `typeNames`
- Replaced deprecated `nme` with `termNames`
- Replaced deprecated `emptyValDef` with `noSelfType`
- In `BoxesRunTime` removed unused method and commented out unused values. Did not delete to keep a reference to the values. If they are deleted people might wonder why `1` and `2` are not used.
- Replaced deprecated `scala.tools.nsc.util.AbstractFileClassLoader` with `scala.reflect.internal.util.AbstractFileClassLoader`

### CODE REVIEW

**Comment by adriaanm** [comment]
Nice reduction in warnings! Diff from the [build of the parent commit](https://scala-webapps.epfl.ch/jenkins/job/pr-scala-publish-core/5204/console):

``` diff
-quick.comp:
-    [mkdir] Created dir: /localhome/jenkins/b/workspace/pr-scala-publish-core/scala/build/quick/classes/compiler
-[quick.compiler] Compiling 314 files to /localhome/jenkins/b/workspace/pr-scala-publish-core/scala/build/quick/classes/compiler
-[quick.compiler] /localhome/jenkins/b/workspace/pr-scala-publish-core/scala/src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala:616: warning: matches with two cases or fewer are emitted using if-then-else instead of switch
-[quick.compiler]               (argsSize : @switch) match {
-[quick.compiler]                            ^
-[quick.compiler] warning: there were 170 deprecation warnings; re-run with -deprecation for details
-[quick.compiler] warning: there were 24 unchecked warnings; re-run with -unchecked for details
-[quick.compiler] warning: there were 266 inliner warnings; re-run with -Yinline-warnings for details
-[quick.compiler] four warnings found
=====================
+quick.comp:
+    [mkdir] Created dir: /home/jenkins/workspace/scala-2.11.x-validate-publish-core/build/quick/classes/compiler
+[quick.compiler] Compiling 314 files to /home/jenkins/workspace/scala-2.11.x-validate-publish-core/build/quick/classes/compiler
+[quick.compiler] warning: there were 64 deprecation warnings; re-run with -deprecation for details
+[quick.compiler] warning: there were 24 unchecked warnings; re-run with -unchecked for details
+[quick.compiler] warning: there were 266 inliner warnings; re-run with -Yinline-warnings for details
+[quick.compiler] three warnings found
```

File: src/compiler/scala/tools/nsc/ast/parser/Parsers.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ import scala.collection.{ mutable, immutable }
 import mutable.{ ListBuffer, StringBuilder }
 import scala.reflect.internal.{ Precedence, ModifierFlags => Flags }
 import scala.reflect.internal.Chars.{ isScalaLetter }
-import scala.reflect.internal.util.{ SourceFile, Position, FreshNameCreator }
+import scala.reflect.internal.util.{ SourceFile, Position, FreshNameCreator, ListOfNil }
```
Comment (Kwestor) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: Why do you need `ListOfNil ` added here?

**Comment by adriaanm** [comment]
Please ignore the  scala-2.11.x-validate-main  -- had to rework the bot's logic. It now only strips the scala-2.11.x- prefix instead of scala-2.11.x-validate-....

**Comment by soc** [comment]
@EECOLOR Not complaining, but: https://github.com/scala/scala/pull/4177?

**Comment by som-snytt** [comment]
@soc That's an interesting collision. It suggests a general rule that removal of API wants to be done in two commits, one to update clients and a second for the API change. Even if the workflow is the opposite order.

**Comment by soc** [comment]
@som-snytt The API changes couldn't be merged, because updating the clients is still pending in https://github.com/scala/scala/pull/4213.

**Comment by EECOLOR** [comment]
@soc I totally missed that. This pull request has a different motivation though. It simply aims at reducing the number of warnings in the scala projects. Yours seems to be focused at working towards actual removal.

Anyway, I wished I had seen it before. I do think however that there is no problem. There will probably be no merge conflicts anyway.

**Comment by soc** [comment]
@EECOLOR No problem. :-)

Let's wait until #4213 gets through, then review/merge #4294 and then I'll add the remaining things from #4177.

**Comment by EECOLOR** [comment]
@soc sounds great!

**Comment by soc** [comment]
Hi @EECOLOR, @adriaanm merged #4213, could you rebase so that the CI runs against it?

**Comment by som-snytt** [comment]
> There will probably be no merge conflicts anyway.

Great minds think alike.

Before I forget: my thought was that if I'm removing deprecated foo in favor of bar, I want two commits, first the removal, then fix the broken (internal) usages in a second commit which can be cherry-picked and also moved ahead of the removal commit.

**Comment by EECOLOR** [comment]
@soc I have rebased it and it seemed to pass the checks

**Comment by soc** [comment]
Most parts look fine to me, with the following issues:
- This is large monolithic commit (not sure if this is an issue, @adriaanm?)
  Maybe we should cherry-pick https://github.com/soc/scala/commit/f1a393641a21a085ec88498724fb8f2384302225 and https://github.com/soc/scala/commit/9257a6c0e72deb7c3ab3c66c6f740002a30918d1, and then apply the rest of this PR to have at least some separation?
- The wholesale conversion from `debugwarn` to `devWarning`. I think this needs more care, because of the intention that dev warnings should be infrequent (https://github.com/scala/scala/blob/2.11.x/src/compiler/scala/tools/nsc/Global.scala#L283) and because the wiring of `isDeveloper` seems to be all over the place. (Compare https://github.com/scala/scala/blob/2.11.x/src/compiler/scala/tools/nsc/Global.scala#L281 with https://github.com/scala/scala/blob/2.11.x/src/reflect/scala/reflect/internal/SymbolTable.scala#L64 for instance)

**Comment by adriaanm** [comment]
Thanks for looking at this, @soc, I agree it would be an improvement to have a small number of more targeted commits, and it also makes sense to consolidate the work you've been doing in this area here.

**Comment by soc** [comment]
Ok, done: https://github.com/soc/scala/compare/scala:2.11.x...soc:topic/remove-deprecation-warnings?expand=1

**Comment by soc** [comment]
Concerning issue 1, I reviewed the changes and it looks like that these specific changes should be fine as a `devWarning`. I'll try building with `-Xdev` to check that it doesn't flood us with warning messages.

**Comment by soc** [comment]
https://github.com/scala/scala/pull/4318

**Comment by EECOLOR** [comment]
So this pull request should be closed right?

**Comment by soc** [comment]
@EECOLOR Yes, let's move further discussion to that PR.

**Pull request closed without merging**